src/pybind/mgr/cephadm/module.py

```diff
@@ -2785,7 +2785,18 @@ def upgrade_status(self):
         return trivial_result(r)
 
     def upgrade_start(self, image, version):
+        if self.mode != 'root':
+            raise OrchestratorError('upgrade is not supported in %s mode' % (
+                self.mode))
         if version:
+            try:
+                (major, minor, patch) = version.split('.')
+                assert int(minor) >= 0
+                assert int(patch) >= 0
```

Comment:
make this less restrictive? currently only major/minor is reported for nfs-ganesha https://github.com/ceph/ceph/pull/33064/commits/d4fabe294f4031d89d2bca067dc6056a4c5f89f7

Reply:
these are only ceph versions